### PR TITLE
add route prefix to webapp

### PR DESCRIPTION
**Pull request by elael**
Explicit the 'route prefix' to the api with 'api/' as default.

### CODE REVIEW

File: lib/roby/interface/rest/server.rb [hunk 1]
```diff
@@ -194,10 +204,10 @@ def server_alive?
                 #
                 # @raise InvalidServer if there is a server at the expected
                 #   host and port, but not a Roby REST server
-                def self.server_alive?(host, port)
+                def self.server_alive?(host, port, main_route)
```
Comment (doudou) [inline lib/roby/interface/rest/server.rb]: Set to '/api' by default here

File: lib/roby/interface/rest/server.rb [hunk 2]
```diff
@@ -18,6 +18,11 @@ class Server
                 # @return [String]
                 attr_reader :host
 
+                # The host the server routes through
+                #
+                # @return [String]
+                attr_reader :main_route
```
Comment (doudou) [inline lib/roby/interface/rest/server.rb]: The documentation is very confusing. I'd prefer `The path under which this API will be available, e.g. /api`

File: lib/roby/interface/rest/server.rb (hunk 1 above)
Comment (elael) [reply lib/roby/interface/rest/server.rb]: I stick to the host/port pattern, where the default is only on the constructor. Sure you want this argument to have a default here? @doudou

File: lib/roby/interface/rest/server.rb (hunk 2 above)
Comment (elael) [reply lib/roby/interface/rest/server.rb]: done

File: lib/roby/interface/rest/server.rb (hunk 1 above)
Comment (doudou) [reply lib/roby/interface/rest/server.rb]: Actually, you've got a point. The port could also have a default.

But more importantly, this breaks backward compatibility for no strong enough reason.

**Comment by elael** [comment]
@doudou  any other changes?

**Pull request merged**